src/main/scala/io/iohk/ethereum/jsonrpc/ExpiringMap.scala

```diff
@@ -9,34 +10,35 @@ import scala.util.Try
 
 object ExpiringMap {
 
- case class ValueWithDuration[V](value: V, expiration: Instant)
+ case class ValueWithDuration[V](value: V, expiration: Duration)
 
-  def empty[K, V](defaultElementRetentionTime: Duration,
-                  clock: Clock = Clock.systemUTC()): ExpiringMap[K, V] =
-    new ExpiringMap(mutable.Map.empty, clock, defaultElementRetentionTime)
+  def empty[K, V](defaultElementRetentionTime: Duration): ExpiringMap[K, V] =
+    new ExpiringMap(mutable.Map.empty, defaultElementRetentionTime)
 }
 /**
   * Simple wrapper around mutable map which enriches each element with expiration time (specified by user or default)
-  * Map is passive which means it only check for expiration and remove expired element during get function
+  * Map is passive which means it only check for expiration and remove expired element during get function.
+  * Duration in all calls is relative to current System.nanoTime()
   */
+//TODO: Make class thread safe
 class ExpiringMap[K, V] private (val underLaying: mutable.Map[K, ValueWithDuration[V]],
-                                 val clock: Clock,
                                  val defaultRetentionTime: Duration) {
+  private val maxHoldDuration = ChronoUnit.CENTURIES.getDuration
 
-  def addUntil(k: K, v: V, until: Instant): ExpiringMap[K, V] = {
-    underLaying += k -> ValueWithDuration(v, until)
+  def addUntil(k: K, v: V, duration: Duration): ExpiringMap[K, V] = {
+    underLaying += k -> ValueWithDuration(v, Try(currentPlus(duration)).getOrElse(currentPlus(maxHoldDuration)))
```

Comment:
What can fail here?

Reply:
for example `Duration.ofNanos(System.nanoTime()).plus(Duration.ofSeconds(Long.MaxValue))` can overflow and give us`java.lang.ArithmeticException: long overflow`